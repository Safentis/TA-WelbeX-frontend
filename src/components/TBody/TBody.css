.tbody {
  background-color: var(--main-grey-color);
}

.tbody__td {
  color: var(--main-dark-blue-color);
  font-size: var(--main-td-size);
  text-align: center;
}
