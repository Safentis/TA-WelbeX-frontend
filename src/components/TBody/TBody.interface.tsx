import { Entry } from '../../application/App/App.interface';

export interface Props {
  filtered: Entry[];
}
