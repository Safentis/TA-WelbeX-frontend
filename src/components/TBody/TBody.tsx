import React, { FC } from 'react';

import './TBody.css';
import { Props } from './TBody.interface';

/**
 * THead is body of table component
 * @returns {React.ReactElement} - returns react element
 */
const TBody: FC<Props> = ({}): React.ReactElement => {
    return (
        <tbody className="tbody">
            <tr className="tbody__tr">
                <td className="tbody__td">
                    2222-22-22 123:33:33
                </td>
                <td className="tbody__td">
                    Cyberunk2077
                </td>
                <td className="tbody__td">
                    12
                </td>
                <td className="tbody__td">
                    22
                </td>
            </tr>
        </tbody>
    );
};

export default TBody;