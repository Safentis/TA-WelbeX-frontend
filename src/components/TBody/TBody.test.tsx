import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import TBody from './TBody';

import { Props } from './TBody.interface';

const setUp = (props: Props) => shallow(<TBody {...props} />);

describe('TBody', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    filtered: [],
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 tbody element', () => {
      expect(component.find('thead')).toBeTruthy();
    });

    it('has 1 tr elemnt', () => {
      expect(component.find('tr')).toBeTruthy();
    });

    it('has 0 td element', () => {
      expect(component.find('td').length === 0).toBeTruthy();
    });
  });
});
