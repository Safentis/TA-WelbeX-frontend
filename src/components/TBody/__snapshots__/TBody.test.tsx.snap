// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TBody component does match to the snapshot 1`] = `
<tbody
  className="tbody"
/>
`;
